index/ask/runner/runner.go

```diff
@@ -0,0 +1,299 @@
+package runner
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/api/apistruct"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/ipfs/go-datastore"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/textileio/powergate/index/ask"
+	"github.com/textileio/powergate/index/ask/internal/metrics"
+	"github.com/textileio/powergate/index/ask/internal/store"
+	"github.com/textileio/powergate/signaler"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+)
+
+var (
+	qaRatelim         = 20
+	qaTimeout         = time.Second * 10
+	qaRefreshInterval = 20 * time.Minute
+
+	log = logging.Logger("index-ask")
+)
+
+// Runner contains cached information about markets
+type Runner struct {
+	api      *apistruct.FullNodeStruct
+	store    *store.Store
```

Comment:
This is a new thing, to separate index persistence and loading on a datastore.
This file was doing too much.